torch/csrc/distributed/rpc/torchscript_functions.cpp

```diff
@@ -64,14 +63,14 @@ c10::intrusive_ptr<c10::ivalue::Future> rpcTorchscript(
   // Create a JIT future and pass it to futMessage's callback to set state
   // of the JIT future.
   auto futPtr = c10::make_intrusive<c10::ivalue::Future>(returnType);
-  std::weak_ptr<FutureMessage> wp = futMessage;
-  futMessage->addCallback(at::wrapPropagateTLSState<void>([futPtr, wp]() {
-    auto futMessage = wp.lock();
-    if (futMessage->hasError()) {
-      c10::ivalue::Future::FutureError jitFutErr(futMessage->error()->what());
-      futPtr->setError(std::make_exception_ptr(jitFutErr));
+  std::weak_ptr<JitFuture> wp = jitFuture;
```

Comment:
Nit: this code is mixing the `c10::ivalue::Future` and the `JitFuture` names, although the latter is just an alias for the former. An unaware reader might get confused by this. Are there plans to later on replace all the JitFuture occurrences with ivalue::Future?

Reply:
Yep, will add a few followup PRs: 1) replace utils/future.h with ivalue::Future in RRef, 2) remove utils/future.h, 3) replace `shared_ptr<JitFuture>` with `intrusive_ptr<JitFuture>`, 4) rename `JitFuture` occurrences to `ivalue::Future`. 

But let me make this PR consistent first.